src/components/Counter.test.js

```diff
@@ -0,0 +1,59 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom'
+import Counter from "./Counter";
+import { act } from 'react-dom/test-utils';
+
+describe("Check Counter element behavior", () => {
+
+    test("Test that component renders initial value provided in props", () => {
+
+        const propValue = 2;
+
+        render(<Counter defaultValue={propValue}/>)
+
+        const propValueElement = screen.getByText(/count/i);
+
+        expect(propValueElement).toBeInTheDocument();
+    })
+
+    test("Check if decrement button decrements the displayed value", async () => {
+        render(<Counter/>)
+
+        const countElement = screen.getByText(/count/i);
+
+        const countValue = countElement.textContent;
+
+        expect(countValue).toBe('Count: 0');
+
+        const buttonDecrement = screen.getByRole('button', { name: /decrement/i });
+        await act(async () => {
```

Comment:
React testing library already integrated act with its APIs. So in most cases, we do not need to wrap `render` and `fireEvent` in `act`. However, if your test still complains about “not wrapped in act(…)”, you might encounter one of these 4 cases described in [this article](https://davidwcai.medium.com/react-testing-library-and-the-not-wrapped-in-act-errors-491a5629193b).